Finish tests for ArgumentOfCorrectType validation

joemcbride opened the pull request:
Github Issue: #10

tlil87 commented:
LGTM :+1:

Would it be an idea to have some tests covering input values passed through operation variables as well? To ensure that they get passed through semantically correct I mean. Or did you already get that covered when you fixed that bug the other day (where the type was incorrectly determined since it came thru input variables instead of a parsed token)?

joemcbride commented:
I added a number of tests when I fixed that, though that wouldn't come through with validation errors.  I think that type of validation should be covered by `VariablesInAllowedPosition`, [found here](https://github.com/graphql/graphql-js/blob/master/src/validation/rules/VariablesInAllowedPosition.js) in the JS project, example tests [found here](https://github.com/graphql/graphql-js/blob/master/src/validation/__tests__/VariablesInAllowedPosition-test.js).  I can slate that one as the next one to tackle, once I finish the `ScalarLeaf` tests.

tlil87 commented:
Sounds good :+1:
